### PR TITLE
Convert pickle macro into regular method.

### PR DESCRIPTION
**Pull request by jsuereth**
* We have to leave around macro implementation for dynamic picklers
* We can only convert `pickleInto` and `pickleTo` for now, as
  the type trickery to do `PickleFormat` lookup and `U <:< PickleFormat.PickleType`
  unification is gunky in regular methods.
* We do not convert `unpickle` for now do to related GPL/runtime lookup
  issues.

Review by @phaller @heathermiller cc> @havocp

### CODE REVIEW

**Comment by jsuereth** [comment]
Hmm, I didn't see this error locally....

File: core/src/main/scala/pickling/AllPicklers.scala [hunk 1]
```diff
@@ -7,9 +7,27 @@ object AllPicklers extends CorePicklersUnpicklers
 
 object all extends CorePicklersUnpicklers {
 
-  implicit class PickleOps[T](picklee: T) {
-    def pickle(implicit format: PickleFormat): format.PickleType = macro Compat.PickleMacros_pickle[T]
-    def pickleInto(builder: PBuilder): Unit = macro Compat.PickleMacros_pickleInto[T]
+  implicit class PickleOps[T: SPickler](picklee: T) {
+    def pickle(implicit format: PickleFormat): format.PickleType = {
```
Comment (havocp) [inline core/src/main/scala/pickling/AllPicklers.scala]: could we also put a `def pickle[T](picklee: T)(implicit pickler: SPickler[T], format: PickleFormat): format.PickleType` into the pickling package object, so you can do `scala.pickling.pickle(t)` without having to use the implicit? Sometimes it's nice to be explicit. Also, you probably conflict with #267 here, I think @eed3si9n changed PickleOps around a bit.

**Comment by phaller** [comment]
Nice! Encouraging that it seems to work well (apart from this one failing test).

**Comment by phaller** [comment]
Without having looked at the code in detail, my guess is that the test failure is due to the missing `clearPicklees()` at the end of the `pickle` extension method. It resets the object IDs that are used when pickling object graphs.

**Comment by jsuereth** [comment]
@phaller see the comment in code about when to do that....  I'm curious when it was emitted.

**Comment by jsuereth** [comment]
Ha, that's actually because I didn't really do anything at all.   There are two different pickleOps methods, so I'm working my way through that now.

**Pull request closed without merging**